tools/rpkg/src/relational.cpp

```diff
@@ -464,10 +464,44 @@ static SEXP result_to_df(duckdb::unique_ptr<QueryResult> res) {
 [[cpp11::register]] SEXP rapi_rel_from_table(duckdb::conn_eptr_t con, const std::string schema_name,
                                              const std::string table_name) {
 	if (!con || !con.get() || !con->conn) {
-		stop("rel_from_df: Invalid connection");
+		stop("rel_from_table: Invalid connection");
 	}
 	auto desc = make_uniq<TableDescription>();
 	auto rel = con->conn->Table(schema_name, table_name);
 	cpp11::writable::list prot = {};
 	return make_external_prot<RelationWrapper>("duckdb_relation", prot, std::move(rel));
 }
+
+[[cpp11::register]] SEXP rapi_rel_from_table_function(duckdb::conn_eptr_t con, const std::string function_name,
+                                                      list positional_parameters_sexps, list named_parameters_sexps) {
+	if (!con || !con.get() || !con->conn) {
+		stop("rel_from_table_function: Invalid connection");
+	}
+	vector<Value> positional_parameters;
+
+	for (sexp parameter_sexp : positional_parameters_sexps) {
+		if (LENGTH(parameter_sexp) != 1) {
+			stop("rel_from_table_function: Need scalar parameter");
+		}
+		positional_parameters.push_back(RApiTypes::SexpToValue(parameter_sexp, 0));
+	}
+
+	named_parameter_map_t named_parameters;
+
+	auto names = named_parameters_sexps.names();
+	if (names.size() != named_parameters_sexps.size()) {
+		stop("rel_from_table_function: Named parameters need names");
+	}
+	idx_t named_parameter_idx = 0;
+	for (sexp parameter_sexp : named_parameters_sexps) {
+		if (LENGTH(parameter_sexp) != 1) {
+			stop("rel_from_table_function: Need scalar parameter");
+		}
+		named_parameters[names[(const R_xlen_t)named_parameter_idx]] = RApiTypes::SexpToValue(parameter_sexp, 0);
+		named_parameter_idx++;
+	}
```

Comment:
Use correct type? Or is there perhaps an idiomatic way to iterate over parallel vectors ("bind" or such)?

```suggestion
	R_xlen_t named_parameter_idx = 0;
	for (sexp parameter_sexp : named_parameters_sexps) {
		if (LENGTH(parameter_sexp) != 1) {
			stop("rel_from_table_function: Need scalar parameter");
		}
		named_parameters[names[named_parameter_idx]] = RApiTypes::SexpToValue(parameter_sexp, 0);
		named_parameter_idx++;
	}
```

Reply:
Better indeed